resteasy-client-reactor-netty/src/test/java/org/jboss/resteasy/client/jaxrs/engines/ReactorNettyClientHttpEngineTest.java

```diff
@@ -490,6 +525,88 @@ public void testThatRequestContentLengthHeaderIsOverwritten() {
     }
 
+    @Test
+    public void testContextInWriter() {
+        final String payload = "hello";
+        final WebTarget target = client.target(url("/echo")).register(new ContextNeedingWriter());
+        final String respBody = target.request().post(Entity.text(payload), String.class);
+        assertEquals(ContextNeedingWriter.transform(payload), respBody);
+    }
+
+    @Test
+    public void testContextInWriterAsyncChaining() throws Exception {
+        final String firstPayload = "first";
+        final String secondPayload = "second";
+        final WebTarget target = client.target(url("/echo")).register(new ContextNeedingWriter());
+
+        final Function<String, CompletionStage<String>> httpGet =
+            payload -> target.request().rx().post(Entity.text(payload), String.class);
+
+        final CompletionStage<String> dataFut = httpGet.apply(firstPayload)
+            .thenCompose(firstResp -> httpGet.apply(secondPayload).thenApply(secResp -> firstResp + " : " + secResp));
+
+        final String data = dataFut.toCompletableFuture().get(500, TimeUnit.MILLISECONDS);
+        assertEquals(
+            ContextNeedingWriter.transform(firstPayload) + " : " + ContextNeedingWriter.transform(secondPayload),
+            data);
+    }
+
+    @Test
+    public void testThatMessageBodyWriterHeadersAreRespected() {
+        final String WRITER_ESTABLISHED_HEADER_VAL = "Binary";
+        class StringWriter implements MessageBodyWriter<String> {
+            @Override
+            public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+                return type == String.class;
+            }
+
+            @Override
+            public long getSize(String s, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+                return s.length();
+            }
+
+            @Override
+            public void writeTo(String s, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream) throws IOException, WebApplicationException {
+                httpHeaders.add("Content-Encoding", WRITER_ESTABLISHED_HEADER_VAL);
+                entityStream.write(s.getBytes());
+            }
+        }
+        final WebTarget target = client.target(url("/headers/content-encoding")).register(new StringWriter());
+        final Response response = target.request().post(Entity.text("marcel sent me some text!"));
+        assertEquals(200, response.getStatus());
+        assertEquals(WRITER_ESTABLISHED_HEADER_VAL, response.readEntity(String.class));
+    }
+
+    @Test
+    public void testCaseInsensitiveHeaderReplace() {
+        final String CONTENT_TYPE_CLIENT_HEADER_VAL = "text/plain";
+        final String CONTENT_TYPE_WRITER_HEADER_VAL = "application/json";
+        final String CONTENT_TYPE_CC_HEADER_NAME = "CONTENT-TYPE";
```

Comment:
LC  stands for Camel Case. What about CC? The string value there certainly isn't Camel Case if that's what it's referring to.

Reply:
I honestly don't remember why I used that naming here. Looking at the code what I wanted to indicate is only lower case and upper case. I will correct these.